src/PrestaShopBundle/Controller/Admin/ProductController.php

```diff
@@ -1310,16 +1311,26 @@ public function searchProductsAction(Request $request): Response
     {
         try {
             $searchPhrase = $request->query->get('search_phrase');
+            $currencyId = $request->query->get('search_currency');
+            $currency = null;
+
+            /** @var CurrencyDataProviderInterface $currencyDataProvider */
+            $currencyDataProvider = $this->container->get('prestashop.adapter.data_provider.currency');
+            if ($currencyId !== null) {
+                $currency = $currencyDataProvider->getCurrencyById((int) $currencyId);
+            } else {
+                $currency = $currencyDataProvider->getCurrencyByIsoCode($currencyDataProvider->getDefaultCurrencyIsoCode());
+            }
 
             /** @var FoundProduct[] $foundProducts */
-            $foundProducts = $this->getQueryBus()->handle(new SearchProducts($searchPhrase, 10));
+            $foundProducts = $this->getQueryBus()->handle(new SearchProducts($searchPhrase, 10, $currency));
 
             $serializer = $this->get('prestashop.bundle.snake_case_serializer_json');
 
-            return new Response($serializer->serialize($foundProducts, 'json'));
+            return new Response($serializer->serialize($foundProducts, 'json'), 200, ['Content-Type' => 'application/json']);
```

Comment:
Maybe use a `JsonResponse` here (no need for serializer not http headers then)

Reply:
Or I believe the `$this->json()` function is simimar

Reply:
:ballot_box_with_check: